File: src/main/java/com/saeed/tasky/models/Project.java
package com.saeed.tasky.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity(name = "projects")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Project {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id")
    private long id;
    @Column(name = "name")
    private String name;
    @Column(name = "icon")
    private String icon;
    @Column(name = "created_at")
    private Timestamp createdAt;
    @Column(name = "modified_at")
    private Timestamp modifiedAt;
}

File: src/main/java/com/saeed/tasky/models/Project.kt
package com.saeed.tasky.models

import javax.persistence.*
import java.sql.Timestamp
import java.time.LocalDateTime

@Entity(name = "projects")
data class Project(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        @Column(name = "id")
        var id: Long? = 0,
        @Column(name = "projectName")
        var name: String? = null,
        @Column(name = "icon")
        var icon: String? = null,
        @Column(name = "created_at")
        var createdAt: LocalDateTime = LocalDateTime.now(),
        @Column(name = "modified_at")
        var modifiedAt: Timestamp? = null
)